Assets/Scripts/App.cs

```diff
@@ -1895,57 +1895,71 @@ private void MoveUserDataFromOldLocation()
 
             if (Directory.Exists(m_UserPath))
             {
-                return;
+                if(File.Exists(Path.Combine(m_UserPath, "MovedFiles.txt")))
```

Comment:
This is a good optimization, but I don't know if we really need it; I think it might also miss an important case (see also the comment below).

If we don't have this logic, and everything goes according to plan, the MoveDirectory() call below will return almost immediately. So we don't gain much.

The only case I can think of where we have files in the old location and the new location and a MovedFiles.txt is if someone loads a newer version (>2.3.22) and then loads an earlier version. In that case, when they go back to 2.3.22+, we should move the files that they have since made, although they may have duplicate names (see below)

Reply:
I agree that it has to be at the end, but I'm afraid it's still problematic because of the downgrade case.